core/src/test/java/io/questdb/griffin/WhereClauseParserTest.java

```diff
@@ -382,6 +382,46 @@ public void testComplexInterval5() throws Exception {
         runWhereTest("timestamp in '2015-02-23T10:00:55.000Z;30m' and timestamp != '2015-02-23T10:10:00.000Z'", "[{lo=2015-02-23T10:00:55.000000Z, hi=2015-02-23T10:09:59.999999Z},{lo=2015-02-23T10:10:00.000001Z, hi=2015-02-23T10:30:55.000000Z}]");
     }
 
+    @Test
+    public void testComplexIntervalNeg1() throws Exception {
+        runWhereTest("timestamp in '2015-02-23T10:00:55;-15s'", "[{lo=2015-02-23T10:00:40.000000Z, hi=2015-02-23T10:00:55.999999Z}]");
+    }
+
+    @Test
+    public void testComplexIntervalNeg2() throws Exception {
+        runWhereTest("timestamp in '2015-02-23T10:00:55;-30m'", "[{lo=2015-02-23T09:30:55.000000Z, hi=2015-02-23T10:00:55.999999Z}]");
+    }
+
+    @Test
+    public void testComplexIntervalNeg3() throws Exception {
+        runWhereTest("timestamp in '2015-02-23T10:00:55;-4h'", "[{lo=2015-02-23T06:00:55.000000Z, hi=2015-02-23T10:00:55.999999Z}]");
+    }
+
+    @Test
+    public void testComplexIntervalNeg4() throws Exception {
+        runWhereTest("timestamp in '2015-02-23T10:00:55;-7d'", "[{lo=2015-02-16T10:00:55.000000Z, hi=2015-02-23T10:00:55.999999Z}]");
+    }
+
+    @Test
+    public void testComplexIntervalNegMs1() throws Exception {
+        runWhereTest("timestamp in '2015-02-23T10:00:55.000Z;-15s'", "[{lo=2015-02-23T10:00:40.000000Z, hi=2015-02-23T10:00:55.000000Z}]");
+    }
+
+    @Test
+    public void testComplexIntervalNegMs2() throws Exception {
+        runWhereTest("timestamp in '2015-02-23T10:00:55.000Z;-30m'", "[{lo=2015-02-23T09:30:55.000000Z, hi=2015-02-23T10:00:55.000000Z}]");
+    }
+
+    @Test
+    public void testComplexIntervalNegMs3() throws Exception {
+        runWhereTest("timestamp in '2015-02-23T10:00:55.000Z;-4h'", "[{lo=2015-02-23T06:00:55.000000Z, hi=2015-02-23T10:00:55.000000Z}]");
+    }
+
+    @Test
+    public void testComplexIntervalNegMs4() throws Exception {
+        runWhereTest("timestamp in '2015-02-23T10:00:55.000Z;-7d'", "[{lo=2015-02-16T10:00:55.000000Z, hi=2015-02-23T10:00:55.000000Z}]");
+    }
+
```

Comment:
Would be good to add tests with 4 token intervals, like 

`'2015-02-23T10:00:55.000Z;-7d;15d;5'`

and

`'2015-02-23T10:00:55.000Z;-7d;-15d;5'`

Both 15d and -15d above doesn't make too much sense to me tbh, perhaps 4 token period with negative intervals should not be valid.